src/azure-cli-core/azure/cli/core/commands/__init__.py

```diff
@@ -272,6 +273,10 @@ def _execute_command(kwargs):
         try:
             op = get_op_handler(operation)
             result = op(client, **kwargs) if client else op(**kwargs)
+
+            if expose_no_wait and kwargs.get('raw', None):
+                return None #return None for 'no-wait'
```

Comment:
Or perhaps it should be an error instead of a warning. If you register no-wait on a command that doesn't support raw, then I would argue it would be better to have that fail positively than just vaguely "not work".